source/Superluminal-RESTfulAPI/LocalMemoryCacheRepository.class.st

```diff
@@ -0,0 +1,53 @@
+Class {
+	#name : #LocalMemoryCacheRepository,
+	#superclass : #CacheRepository,
+	#instVars : [
+		'expiringResources',
+		'timeSourceProvider'
+	],
+	#category : #'Superluminal-RESTfulAPI-Cache'
+}
+
+{ #category : #'instance creation' }
+LocalMemoryCacheRepository class >> using: aTimeSourceProvider [
+	
+	^ self new initializeUsing: aTimeSourceProvider
+]
+
+{ #category : #querying }
+LocalMemoryCacheRepository >> get: aBuiltKey [
+
+	^ expiringResources
+		at: aBuiltKey
+		ifPresent: [ :resource | 
+			( resource isExpiredAt: self now )
+				ifTrue: [ KeyNotFound signal: ( '<1p> not found' expandMacrosWith: aBuiltKey ) ]
+				ifFalse: [ resource ]
+			]
+		ifAbsent: [ KeyNotFound signal: ( '<1p> not found' expandMacrosWith: aBuiltKey ) ]
+]
+
+{ #category : #initialization }
+LocalMemoryCacheRepository >> initializeUsing: aTimeSourceProvider [ 
+ 	
+	expiringResources := Dictionary new.
+	timeSourceProvider := aTimeSourceProvider
+]
+
+{ #category : #'private - accessing' }
+LocalMemoryCacheRepository >> now [
+
+	^ timeSourceProvider value
```

Comment:
if the block provides `now` it's not a timeSourceProvider, but rather a nowProvider or a currentTimeProvider